### PR TITLE
[crossbrowsertesting] Add integration for CrossBrowserTesting

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: vividus-extension-selenium/src/main/resources/properties/defaults/default.properties [hunk 1]
```diff
@@ -21,3 +21,6 @@ proxy.mitm.key-store.type=
 proxy.recording.enabled=false
 proxy.trust-all-servers=false
 proxy.ports=0
+
+cloud-test.passed-status=passed
```
Comment (ikalinin1) [inline vividus-extension-selenium/src/main/resources/properties/defaults/default.properties:25]: Please consider spring configuration as a place for this configuration

File: docs/modules/plugins/nav.adoc [hunk 2]
```diff
@@ -36,3 +36,5 @@
 ** xref:plugin-browser-stack.adoc[Browser Stack]
 ** xref:plugin-lambda-test.adoc[LambdaTest]
 ** xref:plugin-sauce-labs.adoc[Sauce Labs]
+** SmartBear
+*** xref:plugin-cross-browser-testing.adoc[CrossBrowserTesting]
```
Comment (valfirst) [inline docs/modules/plugins/nav.adoc:40]:
```suggestion
** xref:plugin-cross-browser-testing.adoc[SmartBear: CrossBrowserTesting]
```

File: docs/modules/plugins/pages/plugin-cross-browser-testing.adoc [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+----
+configuration.profiles=crossbrowsertesting/web,web/desktop/chrome
+----
+
+[IMPORTANT]
+CrossBrowserTesting profile must be first.
+
+Explore https://support.smartbear.com/crossbrowsertesting/docs/automated-testing/automation-capabilities.html/[Automation Capabilities] for more fine-grained CrossBrowserTesting configuration.
+
+TIP: Links to CrossBrowserTesting test runs can be found in Allure report
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-cross-browser-testing.adoc:38]: it looks like a lot of these content is copy-pasted, doesn't it?

File: vividus-plugin-crossbrowsertesting/build.gradle [hunk 4]
```diff
@@ -0,0 +1,13 @@
+project.description = 'Vividus plugin for CrossBrowserTesting'
+
+dependencies {
+    implementation project(':vividus-bdd-engine')
+    implementation project(':vividus-extension-selenium')
+
+    implementation(group: 'com.github.uarlouski', name: 'cbthelper-java', version: 'ac68dae3ab')
```
Comment (valfirst) [inline vividus-plugin-crossbrowsertesting/build.gradle:7]: please add a link to the branch

File: vividus-plugin-crossbrowsertesting/src/main/java/org/vividus/crossbrowsertesting/CBTTestLinkPublisher.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import com.crossbrowsertesting.AutomatedTest;
+import com.crossbrowsertesting.Builders;
+import com.google.common.eventbus.EventBus;
+import com.mashape.unirest.http.exceptions.UnirestException;
+
+import org.vividus.selenium.IWebDriverProvider;
+import org.vividus.selenium.cloud.AbstractCloudTestLinkPublisher;
+import org.vividus.testcontext.TestContext;
+
+public class CBTTestLinkPublisher extends AbstractCloudTestLinkPublisher
```
Comment (valfirst) [inline vividus-plugin-crossbrowsertesting/src/main/java/org/vividus/crossbrowsertesting/CBTTestLinkPublisher.java:30]:
```suggestion
public class CbtTestLinkPublisher extends AbstractCloudTestLinkPublisher
```

File: vividus-plugin-crossbrowsertesting/src/main/java/org/vividus/crossbrowsertesting/CBTTestStatusManager.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+import com.crossbrowsertesting.AutomatedTest;
+import com.crossbrowsertesting.Builders;
+import com.mashape.unirest.http.exceptions.UnirestException;
+
+import org.vividus.selenium.IWebDriverProvider;
+import org.vividus.selenium.cloud.AbstractCloudTestStatusManager;
+import org.vividus.testcontext.TestContext;
+
+public class CBTTestStatusManager extends AbstractCloudTestStatusManager
```
Comment (valfirst) [inline vividus-plugin-crossbrowsertesting/src/main/java/org/vividus/crossbrowsertesting/CBTTestStatusManager.java:27]:
```suggestion
public class CbtTestStatusManager extends AbstractCloudTestStatusManager
```

File: vividus-plugin-crossbrowsertesting/src/main/java/org/vividus/selenium/crossbrowsertesting/CBTCapabilitiesConfigurer.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+ * limitations under the License.
+ */
+
+package org.vividus.selenium.crossbrowsertesting;
+
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.vividus.bdd.context.IBddRunContext;
+import org.vividus.selenium.AbstractDesiredCapabilitiesConfigurer;
+
+public class CBTCapabilitiesConfigurer extends AbstractDesiredCapabilitiesConfigurer
```
Comment (valfirst) [inline vividus-plugin-crossbrowsertesting/src/main/java/org/vividus/selenium/crossbrowsertesting/CBTCapabilitiesConfigurer.java:23]:
```suggestion
public class CbtCapabilitiesConfigurer extends AbstractDesiredCapabilitiesConfigurer
```

File: vividus-tests/build.gradle [hunk 8]
```diff
@@ -20,6 +20,7 @@ dependencies {
     implementation project(':vividus-plugin-azure-storage-account')
     implementation project(':vividus-plugin-azure-storage-queue')
     implementation project(':vividus-plugin-browserstack')
+    implementation project(':vividus-plugin-crossbrowsertesting')
```
Comment (valfirst) [inline vividus-tests/build.gradle:23]: also please add a dependency to docker

File: vividus-extension-selenium/src/main/java/org/vividus/selenium/cloud/AbstractCloudTestStatusManager.java [hunk 9]
```diff
@@ -32,11 +33,16 @@
 
     private static final Object KEY = CloudTestStatus.class;
 
+    private final String passedStatus;
+    private final String failedStatus;
     private final IWebDriverProvider webDriverProvider;
     private final TestContext testContext;
 
-    protected AbstractCloudTestStatusManager(IWebDriverProvider webDriverProvider, TestContext testContext)
+    protected AbstractCloudTestStatusManager(String passedStatus, String failedStatus,
```
Comment (valfirst) [inline vividus-extension-selenium/src/main/java/org/vividus/selenium/cloud/AbstractCloudTestStatusManager.java:41]: Does it make sense to move `passedStatus` and `failedStatus` to own POJO?

**Review by valfirst — CHANGES_REQUESTED** [review]

**Comment by ikalinin1** [comment]
Please add system test

File: docs/modules/plugins/pages/plugin-cross-browser-testing.adoc (hunk 3 above)
Comment (uarlouski) [reply docs/modules/plugins/pages/plugin-cross-browser-testing.adoc:38]: :)

Comment (uarlouski) [reply docs/modules/plugins/pages/plugin-cross-browser-testing.adoc:38]: it's just a coincidence :)

**Comment by uarlouski** [comment]
> Please add system test

Please provide an account for that

**Review by ikalinin1 — APPROVED** [review]

File: docs/modules/plugins/pages/plugin-cross-browser-testing.adoc [hunk 10]
```diff
@@ -0,0 +1,32 @@
+:grid-default-hostname: hub.crossbrowsertesting.com
+:test-cloud-name: CrossBrowserTesting
+:test-cloud-key: crossbrowsertesting
+:test-cloud-link: https://crossbrowsertesting.com/
+:test-cloud-capabilities: https://support.smartbear.com/crossbrowsertesting/docs/automated-testing/automation-capabilities.html/
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-cross-browser-testing.adoc:5]:
```suggestion
:test-cloud-capabilities: https://support.smartbear.com/crossbrowsertesting/docs/automated-testing/automation-capabilities.html
```

File: vividus-extension-selenium/src/main/java/org/vividus/selenium/cloud/AbstractCloudTestLinkPublisher.java [hunk 11]
```diff
@@ -45,12 +50,13 @@ protected AbstractCloudTestLinkPublisher(String testCloudName, IWebDriverProvide
...
     {
         this.linkName = format("%s Test URL", testCloudName);
+        this.testCloudName = testCloudName;
         this.webDriverProvider = webDriverProvider;
         this.eventBus = eventBus;
         this.testContext = testContext;
     }
 
-    protected abstract Optional<String> getCloudTestUrl(String sessionId);
+    protected abstract Optional<String> getCloudTestUrl(String sessionId) throws GetCloudTestUrlException;
```
Comment (valfirst) [inline vividus-extension-selenium/src/main/java/org/vividus/selenium/cloud/AbstractCloudTestLinkPublisher.java:59]:
```suggestion
    protected abstract String getCloudTestUrl(String sessionId) throws GetCloudTestUrlException;
```

File: vividus-plugin-browserstack/src/main/resources/spring.xml [hunk 12]
```diff
@@ -20,7 +20,14 @@
     <bean id="debugSteps" class="org.vividus.bdd.browserstack.steps.DebugSteps" />
     <bean id="browserStackLinkPublisher" class="org.vividus.browserstack.BrowserStackTestLinkPublisher" />
 
-    <bean class="org.vividus.browserstack.BrowserStackTestStatusManager" lazy-init="false" />
+    <bean class="org.vividus.browserstack.BrowserStackTestStatusManager" lazy-init="false" >
+        <constructor-arg index="0">
+            <bean class="org.vividus.selenium.cloud.CloudTestStatusMapping">
+                <constructor-arg index="0" value="passed" />
+                <constructor-arg index="1" value="failed" />
+            </bean>
```
Comment (valfirst) [inline vividus-plugin-browserstack/src/main/resources/spring.xml:28]: what about moving to Java code?

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

**Pull request merged**